### PR TITLE
Sparse compression

### PR DESCRIPTION
**Pull request by sbrugman**

### CODE REVIEW

**Comment by ieaves** [comment]
Cool extension, are you okay with holding off on this one for a bit. I like your composition idea but want to give it some more thought.

If you wanted you could also refactor that isinstance call into a singledispatch method to beautify that method.

File: src/compressio/compression_algorithms/type_compressions.py [hunk 1]
```diff
@@ -21,6 +22,34 @@ def get_compressed_type(
     return next(test_sequence)
 
 
+def compress_sparse(series: pd.Series) -> pd.Series:
+    number = len(series)
+    missing_count = number - series.count()
+
+    if missing_count > 0:
```
Comment (ieaves) [inline src/compressio/compression_algorithms/type_compressions.py:29]: is this different than `series.hasnans`?

File: src/compressio/type_compressor.py [hunk 2]
```diff
@@ -1,23 +1,51 @@
...
 )
+from compressio.utils import compose
+
+
+@singledispatch
+def parse_func(f):
+    raise NotImplemented
+
+
+@parse_func.register
```
Comment (ieaves) [inline src/compressio/type_compressor.py:32]: This was an issue we hit with visions but parsing the registered dtype from a type annotation wasn't added until 3.7 or 3.8. I think we have to register on the annotation, i.e. `@parse_func.register(object)`

File: src/compressio/compression_algorithms/type_compressions.py (hunk 1 above)
Comment (sbrugman) [reply src/compressio/compression_algorithms/type_compressions.py:29]: We made use of the count before, but now that we don't we're better off with `series.hasnans`

File: src/compressio/type_compressor.py [hunk 3]
```diff
@@ -1,33 +1,29 @@
...
 
 class BaseTypeCompressor:
     def __init__(self, compression_map, *args, **kwargs):
         self.compression_map = compression_map
 
     def compress(self, series: pd.Series, dtype: Type[VisionsBaseType]) -> pd.Series:
-        return self.compression_map.get(dtype, lambda x: x)(series)
+        compression_func = self.compression_map.get(dtype, lambda x: x)
+        if isinstance(compression_func, (list, tuple, set)):
+            compression_func = compose(compression_func)
```
[the remark this answers is not in the document]
Comment (sbrugman) [reply src/compressio/type_compressor.py]: The composition order is from left to right (i.e. `[f(x), g(x), h(x)]` is executed as `f(g(h(x)))`). This order matters (`compression_sparse` should be executed last). It's up to the user to provide a functioning sequence.

File: src/compressio/type_compressor.py [hunk 4]
```diff
@@ -29,13 +29,14 @@ def parse_func(f):
...
+@parse_func.register(object)
 def _(f: object):
     return f
 
 
-@parse_func.register
-def _(f: list):
+@parse_func.register(list)
+@parse_func.register(tuple)
+def _(f: Union[list, tuple]):
```
Comment (ieaves) [inline src/compressio/type_compressor.py:39]: Swanky, I didn't realize this would work!

**Comment by ieaves** [comment]
LGTM, merging

**Pull request merged**